documentation/Generate-code-from-TypeSpec.md

```diff
@@ -0,0 +1,116 @@
+# Before you Start
+
+[TypeScript Azure SDK Design Guidelines](https://azure.github.io/azure-sdk/typescript_introduction.html) is the overall design guideline of the client SDK.
+
+## Prerequisites
+
+- [Node.js 18.x LTS](https://nodejs.org/en/download) or later
+- [Git](https://git-scm.com/downloads)
+
+# Generate SDK
+
+## Use TypeSpec defined in REST API specifications
+
+:ballot_box_with_check: The package structure in the azure-rest-api-specs repository should follow [these guidelines](https://github.com/Azure/azure-rest-api-specs/blob/main/documentation/typespec-structure-guidelines.md#structure).
+
+### How to configure tspconfig.yaml
+You can reference these two config files to configure the Modular or RLC package:
+- [Modular tspconfig.yaml](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/contosowidgetmanager/Contoso.Management/tspconfig.yaml)
+- [RLC tspconfig.yaml](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/ai/Face/tspconfig.yaml)
+
+Please make sure `service-dir`, `package-dir`, `packageDetails`, `isModularLibrary`, `generateMetadata`, `flavor`(for typespec-ts) is correctly configured. `experimentalExtensibleEnums`, `enableOperationGroup`, `hierarchyClient` are the optional configs.
+If you want to enable sample generation with typespec-ts, you should add 
+```
+generateSample:true
+```
+in your tspconfig.yaml
+
+
+- "parameters.service-dir.default" would be `sdk/<service>`
+- "options.@azure-tools/typespec-ts.package-dir" would be `<module>`
+
+SDK module would be generated under the SDK project folder at `sdk/<service>/<module>`.
+
+### Generate Code with code-gen-pipeline tool
+
+Install dependencies to use code-gen-pipeline,  
+```ps
+npm install -g @azure-tools/typespec-client-generator-cli
+npm install -g @microsoft/rush@5.92.0
+npm install -g @azure-tools/js-sdk-release-tools
+```
+
+Create a local json file named generatedInput.json with content similar to that shown below
+```
+   {
+    "dryRun": false,
+    "specFolder": <your-local-spec-repo-path>,
+    "headSha": <commit-id-you-want-to-generate-from>,
+    "repoHttpsUrl": "https://github.com/Azure/azure-rest-api-specs",
+    "relatedTypeSpecProjectFolder": [
+    "specification/SERVICE_DIRECTORY_NAME/PACKAGE_DIRECTORY_NAME/"
+  ]
+}
+```
+
+Run the command
+```
+code-gen-pipeline --inputJsonPath=<path-to-generatedInput.json> --outputJsonPath={path-to-anywhere(This will not be used locally, but this is required to use code-gen-pipeline)}  --typespecEmitter=@azure-tools/typespec-ts  --local 
+```
+
+This command will do:\
+1: generate code with typespec\
+2: generate CHANGELOG.md\
+3: update rush.json(if the generated package is new)\
+4: generate/update ci.mgmt.yml or ci.yml(if the generated package is new)
+
+
+
+### Generate Code with tsp-client tool
```

Comment:
I think we can put it to another doc and call it someting like "advanced way to generate code" since it's only used by a few customer that need to customize their generation